recipes-openxt/openxt/openxt-measuredlaunch/seal-system

```diff
@@ -144,8 +144,17 @@ forward)
 
         # PCR4 is first extended with the digest of EV_SEPARATOR
         # See TCG EFI Protocol Specification 5.2 Crypto Agile Log Entry Format
-        ev_separator="9069ca78e7450a285173431b3e52c5c25299e473"
-        [ "${tpm2}" -eq 0 ] && ev_separator="df3f619804a92fdb4057192dc43dd748ea778adc52bc498ce80524c014b81119"
+        if [ "${tpm2}" -ne 0 ]; then
+            # Read the currently used separator from tpm0 sysfs.
+            if [ -e "/sys/kernel/security/tpm0/ascii_bios_measurements" ]; then
+                ev_separator="$(awk '$1 == 4 && $3 == 04 { print $2 }' /sys/kernel/security/tpm0/ascii_bios_measurements)"
+            else
+                # Pick sha1sum $(printf "\xff\xff\xff\xff" as it looks popular.
+                ev_separator="d9be6524a5f5047db5866813acf3277892a7a30a"
+            fi
+        else
+            ev_separator="df3f619804a92fdb4057192dc43dd748ea778adc52bc498ce80524c014b81119"
+        fi
 
         pcr4=$(hash_extend 0 $ev_separator $hashalg)
```

Comment:
`ev_separator` is unquoted, so this will become `hash_extend 0 [sha1|sha256]` with hash alg defaulting to sha1.  The literal strings sha1 or sha256 will fail the length check and hash_extend will return 1.  `pcr4` then becomes an empty string.

Maybe if ev_separator is unset we should be skipping this extension?  But again, maybe things are too broken to bother with that case.